homeassistant/components/deconz/light.py

```diff
@@ -174,56 +193,58 @@ def brightness(self) -> int | None:
         return self._device.brightness  # type: ignore[no-any-return]
 
     @property
-    def color_temp(self) -> int:
+    def color_temp(self) -> int | None:
         """Return the CT color value."""
         return self._device.color_temp  # type: ignore[no-any-return]
 
     @property
-    def hs_color(self) -> tuple[float, float]:
+    def hs_color(self) -> tuple[float, float] | None:
         """Return the hs color value."""
-        return (self._device.hue / 65535 * 360, self._device.saturation / 255 * 100)
+        if (hue := self._device.hue) and (sat := self._device.saturation):
+            return (hue / 65535 * 360, sat / 255 * 100)
+        return None
 
     @property
     def xy_color(self) -> tuple[float, float] | None:
         """Return the XY color value."""
         return self._device.xy  # type: ignore[no-any-return]
 
     @property
-    def is_on(self) -> bool:
+    def is_on(self) -> bool | None:
         """Return true if light is on."""
         return self._device.state  # type: ignore[no-any-return]
```

Comment:
is the `# type: ignore[no-any-return]` still needed? 
Maybe consider something like this:
```python
if self._device.state is None:
    return None
return bool(self._device.state)   #  check if bool is needed
```

Reply:
All those `# type: ignore[no-any-return]` can be removed with a future release of the library that will be marked with a py.typed file, until then mypy will unfortunately ignore the typing of the library. So I'm edging closer by each PR to publish a new release of the library that can provide typing stubs :)